fpmsyncd/bgpmon.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
```

Comment:
>python [](start = 15, length = 6)

Could you be specific on python2/3?

Reply:
@qiluo-msft This script is not depended on specific version of python2 nor python3.
Do we have to specify with python2 or python3 and can not leave it as "python" if there are no specific dependency?
I did a quick grep from two repositories (sonic-swss, sonic-utilities) and the result seems to be using the same as what I did. only one script did specify python2.  Here is the result in case you are interested:
gechiang@gechiang-vm-01:~/sonic-utilities$ grep -R "/usr/bin/env python" .
./pcieutil/main.py:#!/usr/bin/env python
./sfputil/main.py:#!/usr/bin/env python
./utilities_common/util_base.py:#!/usr/bin/env python2
./scripts/decode-syseeprom:#!/usr/bin/env python
./scripts/dropconfig:#!/usr/bin/env python
./scripts/configlet:#!/usr/bin/env python
./scripts/pfcstat:#!/usr/bin/env python
./scripts/watermarkstat:#!/usr/bin/env python
./scripts/fast-reboot-dump.py:#!/usr/bin/env python
./scripts/queuestat:#!/usr/bin/env python
./scripts/dropstat:#!/usr/bin/env python
./scripts/lldpshow:#!/usr/bin/env python
./scripts/db_migrator.py:#!/usr/bin/env python
./scripts/route_check.py:#!/usr/bin/env python
./scripts/neighbor_advertiser:#!/usr/bin/env python
./scripts/boot_part:#!/usr/bin/env python
./scripts/update_json.py:#! /usr/bin/env python
./scripts/portstat:#!/usr/bin/env python
./scripts/pcmping:#!/usr/bin/env python
./scripts/watermarkcfg:#!/usr/bin/env python
./scripts/intfstat:#!/usr/bin/env python
./scripts/port2alias:#!/usr/bin/env python
./consutil/lib.py:#!/usr/bin/env python
./consutil/main.py:#!/usr/bin/env python
./crm/main.py:#!/usr/bin/env python
./ssdutil/main.py:#!/usr/bin/env python
./acl_loader/main.py:#!/usr/bin/env python
./pddf_psuutil/main.py:#!/usr/bin/env python
./pfc/main.py:#!/usr/bin/env python
./fwutil/lib.py:#!/usr/bin/env python
./fwutil/log.py:#!/usr/bin/env python
./fwutil/main.py:#!/usr/bin/env python
./pddf_fanutil/main.py:#!/usr/bin/env python
./show/mlnx.py:#!/usr/bin/env python
./pddf_ledutil/main.py:#!/usr/bin/env python
./config/nat.py:#!/usr/bin/env python
./config/aaa.py:#!/usr/bin/env python -u
./config/mlnx.py:#!/usr/bin/env python
./psuutil/main.py:#!/usr/bin/env python
./pddf_thermalutil/main.py:#!/usr/bin/env python
./watchdogutil/main.py:#!/usr/bin/env python
gechiang@gechiang-vm-01:~/sonic-utilities$ cd ~/sonic-swss/
gechiang@gechiang-vm-01:~/sonic-swss$ grep -R "/usr/bin/env python" .
./neighsyncd/restore_neighbors.py:#!/usr/bin/env python
./fpmsyncd/bgpmon.py:#!/usr/bin/env python
./fpmsyncd/bgp_eoiu_marker.py:#!/usr/bin/env python
gechiang@gechiang-vm-01:~/sonic-swss$

Let me know if "need to be specific about python2 or python 3" is the standard convention for SONiC?  If so, I can change it to python2.  If not,  will leave it the same as other scripts does.

Reply:
The reason is that python will be symlinked to different executable in different systems, or even does not exist. I assume you tested in python2, so no harm to be specific.

ref: [https://stackoverflow.com/a/19305076](https://stackoverflow.com/a/19305076)


---
In reply to: [484650010](https://github.com/Azure/sonic-swss/pull/1429#discussion_r484650010) [](ancestors = 484650010)

Reply:
Ok. I will set it to python2.